### PR TITLE
CI test (again)

### PR DESCRIPTION
**Pull request by lberk**
/hold

### CODE REVIEW

File: openshift/e2e-tests-openshift.sh [hunk 1]
```diff
@@ -147,7 +147,7 @@ function run_e2e_tests(){
   header "Running tests with Single Tenant Channel Based Broker"
   oc apply -f test/config/st-channel-broker.yaml || return 1
   wait_until_pods_running $EVENTING_NAMESPACE || return 1
-  go_test_e2e -timeout=90m -parallel=12 ./test/e2e -brokerclass=ChannelBasedBroker -channels=messaging.knative.dev/v1alpha1:InMemoryChannel,messaging.knative.dev/v1alpha1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel \
+  go_test_e2e -timeout=90m -parallel=12 ./test/e2e -brokerclass=ChannelBasedBroker -channels=messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel \
```
Comment (matzew) [inline openshift/e2e-tests-openshift.sh:150]: @lberk as discussed on slack,  both brokers on 0.15 have alpha and beta version

we need both, on `Channel...` and on the `MTChannel...`

**Comment by matzew** [comment]
/lgtm

:pray:

File: openshift/e2e-tests-openshift.sh [hunk 2]
```diff
@@ -147,7 +147,7 @@ function run_e2e_tests(){
   header "Running tests with Single Tenant Channel Based Broker"
   oc apply -f test/config/st-channel-broker.yaml || return 1
   wait_until_pods_running $EVENTING_NAMESPACE || return 1
-  go_test_e2e -timeout=90m -parallel=12 ./test/e2e -brokerclass=ChannelBasedBroker -channels=messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel \
+  go_test_e2e -timeout=90m -parallel=12 ./test/e2e -brokerclass=ChannelBasedBroker -channels=messaging.knative.dev/v1alpha1:InMemoryChannel,messaging.knative.dev/v1alpha1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel,messaging.knative.dev/v1beta1:Channel \
```
Comment (matzew) [inline openshift/e2e-tests-openshift.sh:150]: not sure, but it looks like first w/ the IMC ? 

how about the old "ordering" ?
```
-channels=messaging.knative.dev/v1alpha1:Channel,messaging.knative.dev/v1alpha1:InMemoryChannel,messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel \
```

**Comment by matzew** [comment]
Looking at the logs, the test fails for the `ChannelBasedBroker`

@lberk my proposal for debugging would be:

provision an OCP cluster. install pure serving, e.g. via:
* https://github.com/matzew/kn-box/blob/master/01-kn-serving.sh
* apply eventing from this branch, like:
```
ko apply -f config/ && ko apply -f config/channels/in-memory-channel && ko apply -f config/brokers/channel-broker && ko apply -f config/brokers/mt-channel-broker 
```
* run one of the test in question, e.g:
```
go test -v -tags=e2e -count=1 ./test/e2e -brokerclass=ChannelBasedBroker -channels=messaging.knative.dev/v1alpha1:Channel,messaging.knative.dev/v1alpha1:InMemoryChannel,messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel -run ^TestChannelBasedBrokerWithManyTriggers$
```

**Comment by matzew** [comment]
But, I guess the error is due to an incorrect invocation of the test args, see:

https://github.com/lberk/eventing/pull/2

**Comment by matzew** [comment]
I ser test failures in the log, looks like we might need to debug, like:

https://github.com/openshift/knative-eventing/pull/641#issuecomment-644623824

**Comment by lberk** [comment]
> I ser test failures in the log, looks like we might need to debug, like:
> 
> [#641 (comment)](https://github.com/openshift/knative-eventing/pull/641#issuecomment-644623824)

Running things with an OCP cluster, the failing tests pass for me with the method you described.
The test fails seem to be timeouts?

**Comment by lberk** [comment]
@matzew so what's the play here? tests pass on an ocp cluster but not in PR? (where they're just timing out?)

**Comment by matzew** [comment]
lets retest 😂

**Comment by matzew** [comment]
@lberk  I am unable to pull that image:
https://github.com/lberk/eventing/blob/0.15-ci-run/openshift/release/knative-eventing-channelbroker-ci.yaml#L147

hence the `ChannelBroker` based tests fail here

**Comment by matzew** [comment]
@lberk 

Sorry, for this, I forgot about it... and here is what is needed..


## Checkout the `release-v0.15.0` branch and


1. update the `Makefile`

to the `Makefile`, we need to copy the ingress/filter fu (like on 0.14.2)
* https://github.com/openshift/knative-eventing/blob/release-v0.14.2/Makefile#L17-L18
* https://github.com/openshift/knative-eventing/blob/release-v0.14.2/Makefile#L57-L58

Why is that manually? B/c this was removed for `release-next`, since there the images do NOT exist anymore. I simply just forgot it.


2. run `make generate-dockerfiles`, so we get the images added to _THIS_ repos ci-operator

3. run `make update-ci` 


Now PR the changes from 1) and 2) against our release-v0.15.0 branch, and a second PR against the latest master of `openshift/release` (only the "ci-operator/config/openshift/knative-eventing/openshift-knative-eventing-release-v0.15.0" YAML files)



Again, sorry - but that's my fault. I should have created the release branch earlier, b/c now we AGAIN have to backport something that's not on `release-next`, which is the base of cutting releases...

And all this is b/c the number of images on upstream changes... much more than on serving...

File: openshift/release/knative-eventing-channelbroker-ci.yaml [hunk 3]
```diff
@@ -160,3 +160,165 @@ spec:
           containerPort: 9090
         - name: profiling
           containerPort: 8008
+---
```
Comment (matzew) [inline openshift/release/knative-eventing-channelbroker-ci.yaml:163]: I dont think that `make RELEASE...` is needed.

if so, please delete the yamls before (there will be no diff)

File: openshift/release/knative-eventing-channelbroker-ci.yaml [hunk 4]
```diff
@@ -160,3 +160,165 @@ spec:
...
+          - name: CONFIG_OBSERVABILITY_NAME
+            value: config-observability
+          - name: METRICS_DOMAIN
+            value: knative.dev/eventing
+          - name: BROKER_INGRESS_IMAGE
+            value: registry.svc.ci.openshift.org/openshift/knative-v0.15.0:knative-eventing-ingress
+          - name: BROKER_INGRESS_SERVICE_ACCOUNT
+            value: eventing-broker-ingress
+          - name: BROKER_FILTER_IMAGE
+            value: registry.svc.ci.openshift.org/openshift/knative-v0.15.0:knative-eventing-filter
```
Comment (matzew) [inline openshift/release/knative-eventing-channelbroker-ci.yaml:313]: one more manual thing.

looking here: https://github.com/openshift/release/pull/9765/files#diff-627940ffc7041a4e56cb09b23f41c26aR35

these two will be build/pused as:
* `knative-eventing-broker-filter`
* `knative-eventing-broker-ingress`

File: openshift/release/knative-eventing-channelbroker-ci.yaml (hunk 3 above)
Comment (lberk) [reply openshift/release/knative-eventing-channelbroker-ci.yaml]: done

**Pull request merged**